sim/src/make/activity_model.rs

```diff
@@ -2,114 +2,241 @@ use crate::{
     IndividTrip, PersonID, PersonSpec, Scenario, ScenarioGenerator, SpawnTrip, TripEndpoint,
     TripMode,
 };
-use abstutil::{Parallelism, Timer};
+use abstutil::Timer;
 use geom::{Distance, Duration, Time};
-use map_model::{BuildingID, BuildingType, Intersection, Map, PathConstraints, PathRequest};
+use map_model::{BuildingID, BuildingType, Map, PathConstraints, PathRequest};
 use rand::seq::SliceRandom;
 use rand::Rng;
 use rand_xorshift::XorShiftRng;
 
 impl ScenarioGenerator {
     // Designed in https://github.com/dabreegster/abstreet/issues/154
     pub fn proletariat_robot(map: &Map, rng: &mut XorShiftRng, timer: &mut Timer) -> Scenario {
-        let mut residences: Vec<(BuildingID, XorShiftRng)> = Vec::new();
-        let mut workplaces: Vec<BuildingID> = Vec::new();
-        let mut num_residences = 0;
+        let mut residents: Vec<BuildingID> = Vec::new();
+        let mut workers: Vec<BuildingID> = Vec::new();
+
+        let mut num_bldg_residential = 0;
+        let mut num_bldg_commercial = 0;
+        let mut num_bldg_mixed_residential_commercial = 0;
         for b in map.all_buildings() {
             match b.bldg_type {
                 BuildingType::Residential(num_ppl) => {
+                    // TODO is it expected that sometimes num_ppl is 0?
```

Comment:
`classify_bldg` fills this out in the end, and it looks like @matkoniecz deliberately chose 0 sometimes

Reply:
Ah, ok reverting this part for now.